### PR TITLE
New hooks on grid component table

### LINKED ISSUE #22848: Hooks before and after Grid table
#### Is your feature request related to a problem?

Avoid overrides.

#### Describe the solution you'd like

For developers. I want to be able to add dynamic content in selected places.

image

#### Alternatives you've considered

- overrides, but you need to maintain it from version to version
- js inject 🤮 

#### Additional context

Flexibility 🚀 

What about?

`{{ renderhook('displayAdminGridTableBefore', { 'grid': grid }) }}`
`{{ renderhook('displayAdminGridTableAfter', { 'grid': grid }) }}`

cc @PrestaShop/prestashop-maintainers

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | New hooks before and after grid table
| Type?             | new feature
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #22848
| How to test?      | It should be possible to use ps_qualityassurance module and display something in those hook
| Possible impacts? | n/a
| Sponsor company | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/22922)

### CODE REVIEW

**Review by PululuK — DISMISSED** [review]
Nice @kpodemski !
Don't forget to add it here 
https://github.com/PrestaShop/docs/blob/master/src/content/1.7/modules/concepts/hooks/list-of-hooks.md

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig [hunk 1]
```diff
@@ -24,6 +24,14 @@
  *#}
 
 {{ form_start(grid.filter_form, {'attr': {'id': grid.id ~ '_filter_form', 'class': 'table-responsive'}}) }}
+
+{{ renderhook('displayAdminGridTableBefore', {
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig:26]: Should be outside the form. What if you want to include your own form inside it? :thinking:

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig [hunk 2]
```diff
@@ -55,4 +63,12 @@
   </tbody>
   {% block grid_table_footer %}{% endblock %}
 </table>
+
+{{ renderhook('displayAdminGridTableAfter', {
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig:69]: Should be outside the form. What if you want to include your own form inside it? :thinking:

File: src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig (hunk 1 above)
Comment (kpodemski) [reply src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig:26]: Makes sense

Comment (Progi1984) [reply src/PrestaShopBundle/Resources/views/Admin/Common/Grid/Blocks/table.html.twig:26]: :+1: I approve with @PierreRambaud

**Review by Progi1984 — COMMENTED** [review]
Agree with @PierreRambaud

**Comment by kpodemski** [comment]
Thx @Progi1984 @PierreRambaud - ready

**Review by Progi1984 — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]
What a man :muscle:

**Comment by khouloudbelguith** [comment]
Hi @kpodemski,

It is ok :heavy_check_mark: 
image
image

Thank you!

**Comment by PululuK** [comment]
Thanks @kpodemski

**Pull request merged**